secrets/src/int/secret_integers.rs

```diff
@@ -0,0 +1,235 @@
+use super::classify::*;
+use crate::traits::*;
+use std::ops::*;
+
+pub type I8 = Secret<i8>;
+pub type U8 = Secret<u8>;
+pub type I16 = Secret<i16>;
+pub type U16 = Secret<u16>;
+pub type I32 = Secret<i32>;
+pub type U32 = Secret<u32>;
+pub type I64 = Secret<i64>;
+pub type U64 = Secret<u64>;
+pub type I128 = Secret<i128>;
+pub type U128 = Secret<u128>;
+
+// Construct a secret integer
+pub const fn secret<T>(x: T) -> Secret<T> {
+    Secret(x)
+}
+
+impl<T: Scalar> Classify for T {
+    type Classified = Secret<T>;
+    fn classify(self) -> Secret<Self> {
+        secret(self)
+    }
+}
+
+impl<T: Scalar> Declassify for Secret<T> {
+    type Declassified = T;
+    fn declassify(self) -> T {
+        self.0
+    }
+}
+
+impl<'a, T: Scalar> ClassifyRef for &'a T {
+    type ClassifiedRef = &'a Secret<T>;
+    fn classify_ref(self) -> &'a Secret<T> {
+        unsafe { core::mem::transmute(self) }
```

Comment:
Uses of unsafe should always have a SAFETY comment, regardless how obviously correct they are, IMHO

Reply:
I added some comments. Two points:
(1) It should be safe because Secret has a transparent representation
(2) These unsafe functions should never be used in production because they are only enabled when "check-secret-independence" is used.